File: backend/app/models/google_connection.py
from sqlalchemy import Column, String, DateTime, Text, ForeignKey, Boolean
from sqlalchemy.sql import func
from app.core.database import Base
import uuid


class GoogleConnection(Base):
    """Google OAuth connection and tokens"""
    __tablename__ = "google_connections"
    
    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id = Column(String(36), ForeignKey("users.id", ondelete="CASCADE"), nullable=False, unique=True, index=True)
    
    # OAuth data
    connected_email = Column(String(255), nullable=False)
    access_token = Column(Text, nullable=False)  # Encrypted in production
    refresh_token = Column(Text, nullable=False)  # Encrypted in production
    token_expires_at = Column(DateTime(timezone=True), nullable=False)
    
    # Connection status
    is_active = Column(Boolean, default=True, nullable=False)
    
    # Timestamps
    connected_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    last_synced_at = Column(DateTime(timezone=True))
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)
    
    def __repr__(self):
        return f"<GoogleConnection(id={self.id}, email={self.connected_email}, active={self.is_active})>"

File: backend/src/models/GoogleConnection.ts
import { Schema, model, Document, Types } from 'mongoose';

export interface IGoogleConnection extends Document {
  _id: Types.ObjectId;
  userId: Types.ObjectId;
  connectedEmail: string;
  accessToken: string;
  refreshToken: string;
  tokenExpiresAt: Date;
  isActive: boolean;
  connectedAt: Date;
  lastSyncedAt?: Date;
  createdAt: Date;
  updatedAt: Date;
}

const googleConnectionSchema = new Schema<IGoogleConnection>(
  {
    userId: {
      type: Schema.Types.ObjectId,
      ref: 'User',
      required: true,
      unique: true,
      index: true,
    },
    connectedEmail: {
      type: String,
      required: true,
    },
    accessToken: {
      type: String,
      required: true,
    },
    refreshToken: {
      type: String,
      required: true,
    },
    tokenExpiresAt: {
      type: Date,
      required: true,
    },
    isActive: {
      type: Boolean,
      default: true,
    },
    connectedAt: {
      type: Date,
      default: Date.now,
    },
    lastSyncedAt: Date,
  },
  {
    timestamps: true,
  }
);

export const GoogleConnection = model<IGoogleConnection>(
  'GoogleConnection',
  googleConnectionSchema
);
